projects/igniteui-angular/src/lib/grids/grid-mrl-navigation.service.ts

```diff
@@ -0,0 +1,466 @@
+import { Injectable } from '@angular/core';
+import { IgxGridBaseComponent, FilterMode } from './grid-base.component';
+import { first } from 'rxjs/operators';
+import { IgxColumnComponent } from './column.component';
+import { IgxGridGroupByRowComponent } from './grid/groupby-row.component';
+import { IgxGridCellComponent } from './tree-grid';
+import { IgxGridNavigationService } from './grid-navigation.service';
+
+/** @hidden */
+@Injectable()
+export class IgxGridMRLNavigationService extends IgxGridNavigationService {
+    public grid: IgxGridBaseComponent;
+
+    public navigateUp(rowElement, currentRowIndex, visibleColumnIndex, cell?) {
+        if (cell) {
+            this.focusCellUpFromLayout(cell);
+        } else {
+            super.navigateUp(rowElement, currentRowIndex, visibleColumnIndex);
+        }
+    }
+
+    public navigateDown(rowElement, currentRowIndex, visibleColumnIndex, cell?) {
+        if (cell) {
```

Comment:
Why would this function only care about a single parameter. Can't these be integrated better? Maybe extend the SelectionNode interface with information that's needed for MRL so that you don't have to pass a MRL-only flag at the end of all MRL-enabled functions? This ```cell?``` looks very excessive. Besides the other params derive from properties coming from the Cell or its Column component...

Reply:
@ChronosSF The function checks if navigation comes from a cell (could come from something that is not a cell  - for example group row). The cell is also used to get all the mrl specific options - rowStart, colEnd etc.. Extending the SelectionNode  with the mrl data and using just as a single parameter that holds all relevant information is a good idea, however since currently each cell calls directly the navigation service with the related params (rowElement, currentRowIndex, visibleColumnIndex), changing this will require a refactoring in all navigation services and in all components that call them. Since this would be a big change I would prefer we don't do it as part of this PR - rather log a separate issue so that we can work on it separately.